### ISSUE TITLE
Add support for l2ping alike functionality as Low-Level Protocol presence detection

### ISSUE THREAD

**Issue by janlinders** [sputnikdev/eclipse-smarthome-bluetooth-binding#71]

**Comment by janlinders** [comment]
Bluez l2ping utility (device does not need to be in discovery mode as long as the MAC is known)
I.e
Now, lets see how the command responds when the Bluetooth ON

$ sudo l2ping 6C:C4:D5:6C:C5:BC
Ping: 6C:C4:D5:6C:C5:BC from 70:1A:04:59:69:04 (data size 44) ...
44 bytes from 6C:C4:D5:6C:C5:BC id 0 time 5.93ms
44 bytes from 6C:C4:D5:6C:C5:BC id 1 time 8.76ms

Now, lets see how the command responds when the Bluetooth OFF

 $ sudo l2ping 6C:C4:D5:6C:C5:BC
Can't connect: Host is down

**Comment by janlinders** [comment]
Hi xrucka, thanks for the comment. I have the l2ping working through the Exec binding and it works fine. Having similar functionality available in the BT Binding would be much cleaner i guess without sudoers etc.
However if that is becoming to complex, forget the requests.
KR

**Comment by vkolotov** [comment]
Hi @janlinders, could you please elaborate what phone you are using?
As far as I know, the `l2ping` utility is used for older BT devices (prior BT 4.0), new devices that support BT 4.0 won't be able to respond for `l2ping`. It would be great if they could. Please confirm.

Also could you please let us know what is rationale not to have `discoverable` mode enabled on your phone? Is this to achieve privacy?

**Comment by janlinders** [comment]
Hi Vlad, thanks for your reply.

The phones I have available are
Huawei P7-L10 Android 5.1.1 kernel 3.0.8-00595-gc29cecf
IPhone 5S ios 12.0.1
Also my television has BT onboard  : [LG] webOS TV OLED55E6V

All 3 are giving response using l2ping with their respective BT MAC
addresses.

Don't know what BT versions these run. I will see if I can find this.
Discoverable on  means indeed privacy issue and power usage.

I have openhab2 v 2.3 (openhabian) installed on an Raspberry PI Model 2 B+

BT USB dongle : CSR8510 A10

Bluez :● bluetooth.service - Bluetooth service
   Loaded: loaded (/lib/systemd/system/bluetooth.service; enabled; vendor
preset: enabled)
   Active: active (running) since Mon 2018-11-19 20:24:26 CET; 1s ago
     Docs: man:bluetoothd(8)
 Main PID: 21628 (bluetoothd)
   Status: "Running"
   CGroup: /system.slice/bluetooth.service
           └─21628 /usr/libexec/bluetooth/bluetoothd
Nov 19 20:24:26 openHABianPi bluetoothd[21628]: Bluetooth daemon 5.47
Nov 19 20:24:26 openHABianPi bluetoothd[21628]: Starting SDP server
Nov 19 20:24:27 openHABianPi bluetoothd[21628]: Bluetooth management
interface 1.14 initialized

KR, Jan

**Comment by janlinders** [comment]
Found I think the BT versions of the phones.

Huawei P10 BT 4.2, A2DP, LE, aptX
iPhone 5s - BT 4.0 A2DP

**Comment by vkolotov** [comment]
Very interesting. Are you saying that even if your phones are not in discoverable mode (make sure you close BT settings and leave it for 2-3 mins), you still can ping them via `l2ping`?

**Comment by janlinders** [comment]
Confirmed.

**Comment by vkolotov** [comment]
Very good, thank you. As @xrucka mentioned earlier, that would not be very easy to add support for. 

What is rationale not to enable/(force to be enabled) discoverable mode in your devices? Privacy/security?

**Comment by vkolotov** [comment]
And yeah, just double checking. Do you mobile phones have fixed/static MAC address when in discoverable mode or they change it every so often?

**Comment by janlinders** [comment]
Thx.

I'm aware of the tools from the AppStore / PlayStore to force enable
discoverable mode
That means indeed Privacy/Security concerns if I have that on all the time.
So even when from home (anywhere else) people can see the device and it's
vulnerable for attacks.

KR

**Comment by janlinders** [comment]
what i noticed using bluetoothctl (after switching the devices to discover
mode and bluetoothctl scan on) is that they have fixed MAC addresses.

**Comment by vkolotov** [comment]
Right, that's the thing. This is quite unusual for new phones if they have static mac addresses. I'll try to ping my devices to see if it is possible, but I doubt it is... If it is possible, then we might have a good way to track presence, although that would be quite difficult to add support for. Furthermore, looks like bluegiga adapters do not support it at all.

**Comment by janlinders** [comment]
Understood.

Simple to test :

#!/bin/sh
sudo /usr/bin/l2ping -c5 XX:XX:XX:XX:XX:XX >/dev/null 2>&1

if [ $? -eq 0 ]
then
        echo HOME
else
        echo AWAY
fi
